lib/krane/cluster_resource_discovery.rb

```diff
@@ -52,9 +52,16 @@ def fetch_mutating_webhook_configurations
 
     private
 
+    def cluster_url()
```

Comment:
This will prevent us from making a kubectl call each time.
```
def cluster_url
  @cluster_url ||= being 
        raw_json, _, st = kubectl.run("config", "view", "--minify", "--output", "jsonpath={.clusters[*].cluster}", attempts: 5, use_namespace: false)
        st.success? ? JSON.parse(raw_json)["server"] : "/"

  end
end
```

Reply:
Hm interesting, maybe there is a difference between token + cert-based config? This is what i get:

```
~> kubectl config view --minify  --output 'jsonpath={}' --context ci|jq                                                         (ci/test)∞
{
  "apiVersion": "v1",
  "clusters": [
    {
      "cluster": {
        "server": "https://rancher02.xyz.net/k8s/clusters/c-xyz"
      },
      "name": "ci"
    }
  ],
  "contexts": [
    {
      "context": {
        "cluster": "ci",
        "namespace": "test",
        "user": "user-xyz"
      },
      "name": "ci"
    }
  ],
  "current-context": "ci",
  "kind": "Config",
  "preferences": {},
  "users": [
    {
      "name": "user-xyz",
      "user": {
        "token": "REDACTED"
      }
    }
  ]
}
```

Reply:
The jsonpath can/should be further limited to `jsonpath={.clusters[*].cluster.server}`, so we don't pull the cert-data.. that would also eliminate the need to json_parse ?

Reply:
@dturn could you take another look? 
I updated the PR with your suggestions, and modified the retrieval to only getthe server url, not the whole context.
I'm still getting 2 calls to cluster_url, but i guess thats a lot less than before..:

```
[INFO][2021-03-05 09:49:04 +0100]	Discovering resources:
https://rancher02.XYZ.net/k8s/clusters/cxyz
[INFO][2021-03-05 09:49:08 +0100]	  - Deployment/nginx-deployment
[INFO][2021-03-05 09:49:08 +0100]	  - Service/nginx
https://rancher02.xyznet/k8s/clusters/c-xyz
```

regards,
strowi